contrib/submit-protocol/src/App/ProtocolForm.tsx

```diff
@@ -0,0 +1,373 @@
+            </div>
+          </Panel>
+
+          <div className={bootstrapStyles.container}>
+            <div className={bootstrapStyles.modalDialog}>
+              <div className={bootstrapStyles.modalContent}>
+                <div className={bootstrapStyles.modalHeader}>
+                  <h3 className={bootstrapStyles.modalTitle}>
+                    Submit Job v2 <small>Protocol Preview</small>
+                  </h3>
+                </div>
+                <div className={`${bootstrapStyles.modalBody} ${bootstrapStyles.row}`}>
+                  <div className={`${bootstrapStyles.formGroup} ${bootstrapStyles.colMd8}`}>
+                    <TextField
+                      label="Job Name "
+                      value={this.state.jobName}
+                      onChange={this.setJobName}
+                      required={true}
+                    />
+                  </div>
+                  <div className={`${bootstrapStyles.formGroup} ${bootstrapStyles.colMd8}`}>
+                    <Toggle
+                      label="Job Parameters "
+                      checked={this.state.showParameters}
+                      onChange={this.toggleParameters}
+                      inlineLabel={true}
+                    />
+                    {this.renderParameters()}
+                  </div>
+                  <div className={`${bootstrapStyles.formGroup} ${bootstrapStyles.colMd8}`}>
+                    <Label>Protocol YAML Operation</Label>
+                    <label className={bootstrapStyles.colMd3} style={{padding: 0}}>
+                      <a className={`${bootstrapStyles.btn} ${bootstrapStyles.btnSuccess}`}>Import</a>
+                      <input
+                        type="file"
+                        className={bootstrapStyles.srOnly}
+                        accept=".yml,.yaml"
+                        onChange={this.importFile}
+                      />
+                    </label>
+                    <DefaultButton text="View/Edit" onClick={this.openEditor} />
+                  </div>
+                </div>
+                <div className={bootstrapStyles.modalFooter} style={{ marginTop: "150px" }}>
+                  <PrimaryButton text="Submit Job" onClick={this.submitProtocol} />
+                </div>
+              </div>
+            </div>
+          </div>
+
+        </Fabric>
+      </>
+    );
+  }
+
+  private fetchConfig = () => {
+    const source = this.props.source;
+    if (source && source.jobName && source.user) {
+      window.fetch(
+        `${this.props.api}/api/v1/user/${source.user}/jobs/${source.jobName}/config`,
+      ).then((res) => {
+        return res.json();
+      }).then((body) => {
+        const protocol = jsyaml.safeLoad(body);
+        this.setState(
+          { protocol },
+          () => this.setJobName(
+            null as any,
+            `${source.jobName}_clone_${crypto.randomBytes(4).toString("hex")}`,
```

Comment:
Please use Math.random directly since the crypto level random number is not required here, and reduce the bundle size.

Reply:
updated